Backpack/src/androidTest/java/net/skyscanner/backpack/button/BpkButtonLinkTest.kt

```diff
@@ -49,8 +49,8 @@ class BpkButtonLinkTest {
   fun test_icon_end() {
     val trainIcon = AppCompatResources.getDrawable(activity, R.drawable.bpk_train)
     val button = BpkButtonLink(activity).apply {
-      icon = trainIcon
       iconPosition = BpkButton.END
+      icon = trainIcon
```

Comment:
the icon needs to be set after the icon position in tests - this is not the case when running the app, but is due to how the code is updated when no layout was performed like in this test